### PR TITLE
enhancement(bare_metal_server): added support for bare metal server nic ip datasource(s)

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISBareMetalServerNICReservedIPs_basic (24.93s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     26.748s
```
```
--- PASS: TestAccIBMISBareMetalServerNICReservedIP_basic (25.82s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     27.827s
```

### CODE REVIEW

**Comment by ujjwal-ibm** [comment]
moving to upstream

File: website/docs/d/is_bare_metal_server_network_interface_reserved_ip.html.markdown [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+subcategory: "VPC infrastructure"
+layout: "ibm"
+page_title: "IBM : reserved_ip"
+description: |-
+  Shows the info for a reserved IP and bare metal server network interface.
+---
+
+# ibm\_is_bare_metal_server_network_interface_reserved_ip
+
+Import the details of an existing Reserved IP in a network interface of an bare metal server as a read-only data source. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax.
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_bare_metal_server_network_interface_reserved_ip.html.markdown:11]: Link to cloud.ibm.com document page is missing here

File: website/docs/d/is_bare_metal_server_network_interface_reserved_ip.html.markdown [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example Usage
+
+```terraform
+data "ibm_is_bare_metal_server_network_interface_reserved_ip" "data_reserved_ip" {
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_bare_metal_server_network_interface_reserved_ip.html.markdown:16]: data_reserved_ip --> example.. So for every where..  use example

File: website/docs/d/is_bare_metal_server_network_interface_reserved_ips.html.markdown [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+subcategory: "VPC infrastructure"
+layout: "ibm"
+page_title: "IBM : reserved_ips"
+description: |-
+  Lists all the info in reserved IP for Bare Metal Server network interface.
+---
+
+# ibm\_is_bare_metal_server_network_interface_reserved_ips
+
+Import the details of all the Reserved IPs in a network interface of an bare metal server as a read-only data source. You can then reference the fields of the data source in other resources within the same configuration using interpolation syntax.
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_bare_metal_server_network_interface_reserved_ips.html.markdown:11]: Link to cloud.ibm.com document page is missing here

File: website/docs/d/is_bare_metal_server_network_interface_reserved_ips.html.markdown [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+## Example Usage
+
+```terraform
+data "ibm_is_bare_metal_server_network_interface_reserved_ips" "data_reserved_ips" {
```
Comment (SunithaGudisagarIBM) [inline website/docs/d/is_bare_metal_server_network_interface_reserved_ips.html.markdown:16]: use example in all the places

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hi Ujjwal, Please have a look at the PR review comments

**Pull request closed without merging**